geospaas_harvesting/harvesters.py

```diff
@@ -83,24 +83,68 @@ def harvest(self):
                 break
 
 
-class PODAACHarvester(Harvester):
-    """Harvester class for PODAAC data (NASA)"""
+class WebDirectoryHarvester(Harvester):
+    """
+    class for harvesting online data sources that rely on webpages (and most of the time on opendap)
+
+    Child classes should also assign values to the ingester and crawler class attributes.
+    These values should be the crawler and ingester classes to be used in the _create_crawlers()
+    and _create_ingester() methods.
+    """
+    ingester = None
+    crawler = None
+
+    def __init__(self, **config):
+        super().__init__(**config)
+        if 'excludes' in config:
+            if not isinstance(config['excludes'], list):
+                raise HarvesterConfigurationError(
+                    "'excludes' field must be fed with a python list of excluded names ")
```

Comment:
You need to stop adding type checks everywhere

Reply:
https://github.com/nansencenter/django-geo-spaas-harvesting/pull/20#discussion_r474416025